### DISCUSSION TITLE
Should I use a CDN?

### DISCUSSION THREAD

**Discussion by sibrash4** [jpanther/congo#884] [Q&A]
I use Amazon CloudFront for most images except the thumbnails and the featured images, because they require a specific project structure. Sometimes, I've used raw HTML to pass the images instead. However, I deploy my website on Cloudflare, and it made me wonder: since Cloudflare is similar to a CDN, does that already solve the problem since it is already being served through Cloudflare's CDN? Does the same apply for the rest of the images on the website? Should I just migrate all of the website’s images to Cloudflare?

Or am I missing something?

**Comment by tomy0000000** [comment]
Most of your inferences are correct. If you have Cloudflare do all the CDN for you, then you wouldn't need another layer of CDN on CloudFront. In some rare cases, having two layers of CDNs can cause unexpected performance degradation and hard-to-debug issues.

But in reality, it's rare, and visitors won't even notice the differences for most small personal blogs.

**Reply by sibrash4** [reply]
Thanks, that makes sense! It helps clear things up for me.
